### PR TITLE
Major revision of documentation

### LINKED ISSUE #235: Revise documentation
- Address questions and comments from forum, mailing list and face-to-face discussions
- Describe event aggregation options (same log, different logs)
- Describe event bus and event collaboration functionality
- ...

### PR DESCRIPTION
**Pull request by krasserm**
- Stronger focus on event collaboration
- Affects abstract, introduction and architecture (+ figures)
- More consistent documentation of event collaboration
- FAQ entry: how can Kafka be integrated with Eventuate
- closes #235

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,23 +5,22 @@
...
-- derives current application state from logged events (event sourcing)
-- replicates application state by replicating events across multiple locations
-- allows updates to replicated state at multiple locations concurrently (multi-master)
-- allows individual locations to continue writing even if they are partitioned from other locations
-- provides means to detect, track and resolve conflicting updates (interactive and automated)
-- enables applications to implement a causal consistency model
-- provides implementations of operation-based CRDTs
-- supports distribution up to global scale
+Eventuate is a toolkit for building applications composed of event-driven and event-sourced services that collaborate by exchanging events over shared event logs. Services can either be co-located on a single node or distributed up to global scale. Services can also be replicated with causal consistency and remain available for writes during network partitions. Eventuate is written in [Scala](http://www.scala-lang.org/) and built on top of [Akka](http://akka.io), a toolkit for building highly concurrent, distributed, and resilient message-driven applications on the JVM. Eventuate
```
Comment (volkerstampa) [inline README.md]: Maybe add that it comes with a scala and java API?!?

Comment (krasserm) [inline README.md]: Absolutely, good idea!

**Comment by volkerstampa** [comment]
Great improvement, really like the new doc.

**Comment by krasserm** [comment]
@volkerstampa I'm very glad to hear that, thank you!

File: src/sphinx/overview.rst [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+- :ref:`Event-sourced actor <event-sourced-actors>`. An Akka actor that consumes events from an event log and produces events to the same event log. Internal state derived from consumed events is an in-memory *write model* contributing to the command-side (C) of a CQRS_-based application. 
+- :ref:`Event-sourced view <event-sourced-views>`. An Akka actor that only consumes events from an event log. Internal state derived from consumed events is an in-memory *read model* contributing to the query-side (Q) of CQRS.
+- :ref:`Event-sourced writer <event-sourced-writers>`. An Akka actor that consumes events from an event log to update a persistent query database. State derived from consumed events is a persistent read model contributing to the query-side (Q) of CQRS.
```
Comment (cstub) [inline src/sphinx/overview.rst]: shouldn't `read-model` also be italic to align with the previous statements?

File: README.md [hunk 3]
```diff
@@ -5,23 +5,22 @@
...
-- allows individual locations to continue writing even if they are partitioned from other locations
-- provides means to detect, track and resolve conflicting updates (interactive and automated)
-- enables applications to implement a causal consistency model
-- provides implementations of operation-based CRDTs
-- supports distribution up to global scale
+Eventuate is a toolkit for building applications composed of event-driven and event-sourced services that collaborate by exchanging events over shared event logs. Services can either be co-located on a single node or distributed up to global scale. Services can also be replicated with causal consistency and remain available for writes during network partitions. Eventuate has a [Java](http://www.oracle.com/technetwork/java/javase/overview/index.html) and [Scala](http://www.scala-lang.org/) API, is written in Scala and built on top of [Akka](http://akka.io), a toolkit for building highly concurrent, distributed, and resilient message-driven applications on the JVM. Eventuate
+
+- provides event-sourcing abstractions for building stateful services on the command-side and query-side of CQRS-based applications
```
Comment (cstub) [inline README.md]: maybe add a link for CQRS here

File: src/sphinx/index.rst [hunk 4]
```diff
@@ -2,22 +2,21 @@
 Eventuate
 =========
 
-Eventuate is a toolkit for building distributed, highly-available and partition-tolerant event-sourced applications. It is written in Scala_ and built on top of Akka_, a toolkit for building highly concurrent, distributed, and resilient message-driven applications on the JVM. Eventuate
+Eventuate is a toolkit for building applications composed of event-driven and event-sourced services that collaborate by exchanging events over shared event logs. Services can either be co-located on a single node or distributed up to global scale. Services can also be replicated with causal consistency and remain available for writes during network partitions. Eventuate has a Java_ and Scala_ API, is written in Scala and built on top of `Akka`_, a toolkit for building highly concurrent, distributed, and resilient message-driven applications on the JVM. Eventuate
 
+- provides event-sourcing abstractions for building stateful services on the command-side and query-side of CQRS-based applications
```
Comment (cstub) [inline src/sphinx/index.rst]: maybe add a link for CQRS here

**Comment by cstub** [comment]
LGTM. This is a great improvement of the existing documentation, that clarifies the intend and possibilities of Eventuate - i really like it!

**Comment by krasserm** [comment]
@cstub thanks for the review, glad you like the revised documentation! Will apply the proposed changes.

**Pull request merged**